mysql/mysqldump_test.go

```diff
@@ -0,0 +1,561 @@
+				"te_s_t": ddl.CreateTable{
+					Name:     "te_s_t",
+					ColNames: []string{"a__", "b_b", "n_n"},
+					ColDefs: map[string]ddl.ColumnDef{
+						"a__": ddl.ColumnDef{Name: "a__", T: ddl.String{Len: ddl.MaxLength{}}, NotNull: true},
+						"b_b": ddl.ColumnDef{Name: "b_b", T: ddl.String{Len: ddl.MaxLength{}}},
+						"n_n": ddl.ColumnDef{Name: "n_n", T: ddl.Int64{}},
+					},
+					Pks: []ddl.IndexKey{ddl.IndexKey{Col: "a__"}}}},
+		},
+		// The "Data conversion: ..." cases check data conversion for each type.
+		// test remains for binary type
+		{
+			name: "Data conversion: bool, bigint, char, blob",
+			input: `
+	CREATE TABLE test (id integer PRIMARY KEY, a bool, b bigint, c char(1),d blob);
+	INSERT INTO test (id, a, b, c, d) VALUES (1, 1, 42, 'x',_binary '` + string([]byte{137, 80}) + `');`,
+			expectedData: []spannerData{
+				spannerData{table: "test", cols: []string{"id", "a", "b", "c", "d"}, vals: []interface{}{int64(1), true, int64(42), "x", []byte{0x89, 0x50}}}},
+		},
+		{
+			name: "Data conversion: date, float, decimal, mediumint",
+			input: `
+	CREATE TABLE test (id integer PRIMARY KEY, a date, b float, c decimal(3,5));
+	INSERT INTO test (id, a, b, c) VALUES (1,'2019-10-29',4.444,5.44444);
+	`,
+			expectedData: []spannerData{
+				spannerData{table: "test", cols: []string{"id", "a", "b", "c"}, vals: []interface{}{int64(1), getDate("2019-10-29"), float64(4.444), float64(5.44444)}}},
+		},
+		{
+			name: "Data conversion: smallint, mediumint, bigint, double",
+			input: `
+	CREATE TABLE test (id integer PRIMARY KEY, a smallint, b mediumint, c bigint, d double);
+	INSERT INTO test (id, a, b, c, d) VALUES (1, 88, 44, 22, 444.9876);
+	`,
+			expectedData: []spannerData{
+				spannerData{table: "test", cols: []string{"id", "a", "b", "c", "d"}, vals: []interface{}{int64(1), int64(88), int64(44), int64(22), float64(444.9876)}}},
+		},
+		// test with different timezone
+		{
+			name: "Data conversion:  text, timestamp, datetime, varchar",
+			input: `
+	SET TIME_ZONE='+02:30';
+	CREATE TABLE test (id integer PRIMARY KEY, a text, b timestamp, c datetime, d varchar(15));
+	INSERT INTO test (id, a, b, c, d) VALUES (1, 'my text', '2019-10-29 05:30:00', '2019-10-29 05:30:00', 'my varchar');
+	`,
+			expectedData: []spannerData{
+				spannerData{table: "test", cols: []string{"id", "a", "b", "c", "d"}, vals: []interface{}{int64(1), "my text", getTime(t, "2019-10-29T05:30:00+02:30"), getTimeWithoutTimezone(t, "2019-10-29 05:30:00"), "my varchar"}}},
+		},
+	}
+	for _, tc := range multiColTests {
+		conv, rows := runProcessMySQLDump(tc.input)
+		if !tc.expectIssues {
+			noIssues(conv, t, tc.name)
+		}
+		if tc.expectedSchema != nil {
+			assert.Equal(t, tc.expectedSchema, stripSchemaComments(conv.SpSchema), tc.name+": Schema did not match")
+		}
+		if tc.expectedData != nil {
+			assert.Equal(t, tc.expectedData, rows, tc.name+": Data rows did not match")
+		}
+	}
+
+	{ // Test set timezone statement.
```

Comment:
What is the purpose to use `{}` to include the following code?

Reply:
https://github.com/cloudspannerecosystem/harbourbridge/blob/fa3d015562a86e51157a983f900b07a1fb5ca62d/mysql/mysqldump_test.go#L387-L391
Made separate test for this.